### ISSUE TITLE
[FR]: Make freeze logic on install more convenient to end users

### ISSUE THREAD

**Issue by Aghassi** [aspect-build/rules_js#847]
### What is the current behavior?

Currently, a user must export `ASPECT_RULES_JS_FROZEN_PNPM_LOCK` to enable `update_pnpm_lock` failures on CI for non frozen installs.

### Describe the feature

Instead of having the users of this rule know a magic variable, just check for a common one like `CI` so that this works out of box without confusing the end user. A great package for simplifying this check is https://www.npmjs.com/package/env-ci which will do all the necessary checks for you.

### Fund our work

- [ ] Sponsor our open source work by donating a [feature bounty](https://opencollective.com/aspect-build/)

**Comment by arrdem** [comment]
I don't think this is something we can reasonably do since it would create a default behavior with a high likelihood of divergence between "CI" and "local" environments.

Will discuss next steps with the team when the SME is back.
